docs/proposals/service-catalog.md

```diff
@@ -0,0 +1,143 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest release of this document can be found
+[here](http://releases.k8s.io/release-1.1/docs/proposals/service-catalog.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Abstract
+
+Kubernetes has Services but they are scoped per namespace, and (assuming eventual support for RBAC) you may not want everyone to be able to see and use all the services in your namespace.
+
+This proposal is about:
+
+- associating additional configuration/secret data with Services
+- adding a new Service Catalog resource into which users can publish services that others can easily find and consume
+- adding support for dynamic provisioning of resources from the catalog
+- making it easier to link Services (and their configuration data) to Pods
+
+# Use cases
+
+## Shared development database
+
+A development team is working on an application that uses a database. The IT department manages the database (i.e., it's lives off-cluster). All developers share the same credentials to access the database, but these credentials are managed by IT. Rather than having each developer create his or her own `Service` and `Secret` to connect to the database, IT creates a "db-app-xyz" `Service` and a "db-app-xyz" `Secret` in the "info-tech" namespace. IT also has lots of other `Service` resources in their namespace and they don't want to expose all of them to the development team. Therefore, they publish "db-app-xyz" to a service catalog. To use this service, a developer searches for it in the service catalog and adds it to their namespace.
+
+## Dynamic provisioning of cluster resources
+
+A user creates a "template" (e.g. if something similar to [OpenShift Templates](https://docs.openshift.org/latest/dev_guide/templates.html) exists) that makes it easy to create everything needed to spin up a new PostgreSQL database (customizable username/password, Service, Deployment, etc.). The user wants to share only this template in the service catalog so others can find it and use it, while keeping other templates in the namespace private.
+
+
+## Custom dynamic provisioning
```

Comment:
Understood. As mentioned elsewhere I'd prefer we remove provisioning (for now) but keep some concept of binding.